src/container/intrusive_list.cpp

```diff
@@ -0,0 +1,4 @@
+#include "intrusive_list.hpp"
+
+#include <vector>
+namespace yaclib::container {}
```

Comment:
- [ ] Make IntrusiveList better

Since IntrusiveList is used only inside our project with non-statically polymorphic types. 
We know all the types with which it is used.
So write the list methods inside the .cpp file and also declare the required template types in it:
```
template class IntrusiveList<ITask>;
template class IntrusiveList<IThread>;
```